src/Symfony/Component/OptionsResolver/Debug/OptionsResolverIntrospector.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver\Debug;
+
+use Symfony\Component\OptionsResolver\Exception\NoConfigurationException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @final
+ */
+class OptionsResolverIntrospector
```

Comment:
What do you think about move here the inner implementation of `OptionsResolverWrapper`? making constructor argument optional?

Reply:
```
$this->optionsResolver = $optionsResolver ?: $this;
```

Reply:
`OptionsResolverWrapper` does have serve the same purpose and aims to replace an `OptionResolver` instance to not fail on undefined options while configuring it, whereas the introspector just expose an API to get the state of an OptionResolver once configured. Not sure it'll be worth merging the two of them into a `DebugOptionResolver`.

Reply:
fair enough :(